conan/tools/cmake/cmakedeps2/target_configuration.py

```diff
@@ -87,7 +87,7 @@ def _requires(self, info, components):
                     dep_target = self._cmakedeps.get_property("cmake_target_name", dep, comp)
                     dep_target = dep_target or default_target
                     link = not (pkg_type is PackageType.SHARED and
-                                dep_comp.package_type is PackageType.SHARED)
+                                dep_comp.type is PackageType.SHARED)
```

Comment:
Not sure to understand this part of the code neither if using dep_comp here is not particularly risky since it might be None regarding the test above. 

How this should be fixed ?

Reply:
setting link per conditional branches ?